quickwit-config/src/source_config.rs

```diff
@@ -30,9 +30,25 @@ use crate::{is_false, validate_identifier};
 /// Reserved source ID for the `quickwit index ingest` CLI command.
 pub const CLI_INGEST_SOURCE_ID: &str = ".cli-ingest-source";
 
+fn default_num_pipelines() -> usize {
+    1
+}
+
+fn is_one(num: &usize) -> bool {
+    *num == 1
+}
+
 #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
 pub struct SourceConfig {
     pub source_id: String,
+
+    #[doc(hidden)]
+    #[serde(default = "default_num_pipelines", skip_serializing_if = "is_one")]
+    /// Number of indexing pipelines spawned for the source on each indexer.
+    /// Therefore, there will exist `num_pipelines` * `num_indexers` indexing pipelines for the
```

Comment:
(very small detail)

It makes you think that there is a variable `num_indexers` used somewhere.

Therefore, if you start `n` indexers, there will exist `num_pipelines` * `n` indexing pipelines for the source.

Reply:
Good point.